tests/test_binexport_features.py

```diff
@@ -0,0 +1,458 @@
+# Copyright (C) 2024 Mandiant, Inc. All Rights Reserved.
+# Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at: [package root]/LICENSE.txt
+# Unless required by applicable law or agreed to in writing, software distributed under the License
+#  is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and limitations under the License.
+import binascii
+
+import pytest
+import fixtures
+
+import capa.features.file
+import capa.features.insn
+import capa.features.common
+import capa.features.basicblock
+from capa.features.common import (
+    OS,
+    OS_LINUX,
+    ARCH_I386,
+    FORMAT_PE,
+    ARCH_AMD64,
+    FORMAT_ELF,
+    OS_ANDROID,
+    OS_WINDOWS,
+    ARCH_AARCH64,
+    Arch,
+    Format,
+)
+
+FEATURE_PRESENCE_TESTS_BE2_ELF_AARCH64 = sorted(
+    [
+        # file/string
+        (
+            "687e79.be2",
+            "file",
+            capa.features.common.String("AppDataService start"),
+            True,
+        ),
+        ("687e79.be2", "file", capa.features.common.String("nope"), False),
+        # file/sections
+        ("687e79.be2", "file", capa.features.file.Section(".text"), True),
+        ("687e79.be2", "file", capa.features.file.Section(".nope"), False),
+        # file/exports
+        (
+            "687e79.be2",
+            "file",
+            capa.features.file.Export("android::clearDir"),
+            "xfail: not implemented yet?!",
+        ),
+        ("687e79.be2", "file", capa.features.file.Export("nope"), False),
+        # file/imports
+        ("687e79.be2", "file", capa.features.file.Import("fopen"), True),
+        ("687e79.be2", "file", capa.features.file.Import("exit"), True),
+        (
+            "687e79.be2",
+            "file",
+            capa.features.file.Import("_ZN7android10IInterfaceD0Ev"),
+            True,
+        ),
+        ("687e79.be2", "file", capa.features.file.Import("nope"), False),
+        # function/characteristic(loop)
+        (
+            "687e79.be2",
+            "function=0x1056c0",
+            capa.features.common.Characteristic("loop"),
+            "xfail: not implemented yet",
+        ),  # TODO(mr): https://github.com/mandiant/capa/issues/2101
+        (
+            "687e79.be2",
+            "function=0x1075c0",
+            capa.features.common.Characteristic("loop"),
+            False,
+        ),
+        # bb/characteristic(tight loop)
+        (
+            "687e79.be2",
+            "function=0x0",
+            capa.features.common.Characteristic("tight loop"),
+            "xfail: not implemented yet",
+        ),
+        (
+            "687e79.be2",
+            "function=0x0",
+            capa.features.common.Characteristic("tight loop"),
+            "xfail: not implemented yet",
+        ),
```

Comment:
@mr-tz tight loop is implemented - is this a bug?

Reply:
it doesn't work yet for this sample - I hope it's an easy fix in the capa basic block extractor code

Reply:
https://github.com/mandiant/capa/issues/2113

Reply:
@mr-tz I started looking at this - do you have an example address that does not work? I think I found at least one example that's affected by https://github.com/google/binexport/blob/7bba64aa8097d0a1f20a1b3ebbaed7680fc56b76/java/src/main/java/com/google/security/binexport/BinExport2Builder.java#L541-L542